test/mongodb.test.js

```diff
@@ -3372,6 +3372,143 @@ describe('mongodb connector', function() {
     });
   });
 
+  describe('Test processMongoDBUrl function', function() {
+    const module = require('../');
+
+    context('should process mongodb url for default database manipulation', function() {
+      it('when no seetings db, but url has db, multiple hosts, user credentials and options', function() {
+        const url = 'mongodb://userid:password@db1.example.com:27017,db2.example.com:32667/testdb?authSource=admin&replicaSet=replset&readPreference=primary&ssl=true';
+        const database = '';
+        module.processMongoDBURL('', url).should.be.eql('mongodb://userid:password@db1.example.com:27017,db2.example.com:32667/testdb?authSource=admin&replicaSet=replset&readPreference=primary&ssl=true');
```

Comment:
Could we add one test where the URL contains a trailing forward slash without a specified database (e.g. `mongodb://userid:password@db1.example.com/,mongodb://userid:password@db1.example.com/testdb`

Reply:
I already have this test case (trailing forward slash without specified database) already:
```
it("when settings has db, url doesn't have db, has slash, multiple hosts, user creds and no options", function() {
        const url = 'mongodb://userid:password@db1.example.com:27017,db2.example.com:32667/';
        const database = 'mydb';
        module.processMongoDBURL(database, url).should.be.eql('mongodb://userid:password@db1.example.com:27017,db2.example.com:32667/mydb');
      });
```
and with a database
```
it('when settings has db, url too has db, multiple hosts, user credentials and no options', function() {
        const url = 'mongodb://userid:password@db1.example.com:27017,db2.example.com:32667/testdb';
        const database = 'mydb';
        module.processMongoDBURL(database, url).should.be.eql('mongodb://userid:password@db1.example.com:27017,db2.example.com:32667/testdb');
      });
```

Reply:
Ah, thanks! Didn't notice that test.